fluent/fluent.go

```diff
@@ -93,7 +95,20 @@ type Fluent struct {
 }
 
 // New creates a new Logger.
-func New(config Config) (f *Fluent, err error) {
+func New(config Config) (*Fluent, error) {
+	if config.Timeout == 0 {
+		config.Timeout = defaultTimeout
+	}
```

Comment:
Why not having all `if config.ParameteName ...` clauses here, instead of `newWithDialer`?
Those configuration defaults are of Logger, and there seems no strong reason to do it in `newWithDialer`.
Having all the similar things in the same place should be valuable for readability.

Reply:
Ah, code in test cases are assuming that defaults are set in `newWithDialer`. Hmm.

Reply:
Indeed, I moved most of parameters initializations to `newWithDialer()` because I didn't want to refactor test cases too much. However, the timeout is needed only when the dialer is created by `New()`.